hw6/src/lib.rs

```diff
@@ -0,0 +1,195 @@
+pub enum Item {
+    First,
+    Second,
+    Third,
+}
+
+impl Item {
+    pub fn index(&self) -> usize {
+        match self {
+            Item::First => 0,
+            Item::Second => 1,
+            Item::Third => 2,
+        }
+    }
+}
+
+pub struct Tuple(u32, f32, f64);
+pub struct Array([f64; 3]);
+
+trait Indexed {
+    fn get_item(&self, item: Item) -> f64;
+    fn set_item(&mut self, item: Item, value: f64);
+}
+
+trait Container3Elements {
+    fn default_values() -> Self
+    where
+        Self: Sized;
+    fn is_default(&self) -> bool {
+        self.get_item(Item::First) == Default::default()
+            && self.get_item(Item::Second) == Default::default()
+            && self.get_item(Item::Third) == Default::default()
+    }
```

Comment:
Предлагаю заменить на:
```suggestion
    fn is_default(&self) -> bool {
        self == Default::default()
    }
```

Reply:
Можно, но оператор эквавалетности заставит нас "подвезти" еще   
```rust
trait Container3Elements: Default + PartialEq<Self>  
```
и заставить наследников опредедить метод сравнения
``` rust
    fn eq(&self, other: &Self) -> bool {
```
И получим дублирование кода по иерархии.

Reply:
Действительно, согласен. В таком случае оба подхода мне кажутся справедливыми без перевесов в ту или иную сторону

Reply:
С точки зрения дизайна языка проблем нет. С дизайна проектироваиня - идет дублирование кодов наследниках , и при большой иерархии это  может  быть избыточным.

Но Архитектура, это компромисы  - так что это всго лишь  вопрос выбора.